### ISSUE TITLE
Install Request: STAR-related RNA-seq pipeline [IN:04680117]

### ISSUE THREAD

**Issue by heatherkellyucl** [UCL-RITS/rcps-buildscripts#444]
### Arriba 2.1.0
Fast and accurate gene fusion detection from RNA-Seq data
https://github.com/suhrig/arriba
https://arriba.readthedocs.io/en/latest/quickstart/#manual-installation
License: MIT/Expat and GNU GPL v3

Requires: STAR version >=2.7.6a recommended
**(We have `star/2.7.3a` at present).**

### STAR-SEQR
RNA Fusion Detection and Quantification using STAR.
https://github.com/ExpressionAnalysis/STAR-SEQR
License: non-commercial use

Additional Requirements:

- biobambam2(https://github.com/gt1/biobambam2) or conda install -c bioconda biobambam **(newer development is at https://gitlab.com/german.tischler/biobambam2 GPLv3)**
- STAR(https://github.com/alexdobin/STAR). Must use >2.5.3a. conda install -c bioconda star **(see above)**
- Velvet(https://github.com/dzerbino/velvet) or conda install -c bioconda velvet 
**(we have `velvet/1.2.10` which is current)**
- samtools(https://github.com/samtools/samtools) or conda install -c bioconda samtools **(we have `samtools/1.11/gnu-4.9.2` as our newest)**
- Salmon(https://combine-lab.github.io/salmon/) or conda install -c bioconda salmon **(GPLv3)**
- UCSC utils(http://hgdownload.soe.ucsc.edu/admin/exe/) or conda install -c bioconda ucsc-gtftogenepred **(binaries, free for academic, nonprofit, and personal use.)**
- gffread(http://ccb.jhu.edu/software/stringtie/dl/gffread-0.9.8c.tar.gz) or conda install -c bioconda gffread **(http://ccb.jhu.edu/software/stringtie/gff.shtml#gffread)**

### STAR-Fusion 1.10.0
STAR-Fusion uses the STAR aligner to identify candidate fusion transcripts supported by Illumina reads.
https://github.com/STAR-Fusion/STAR-Fusion/wiki
License: BSD-3-Clause

Requires:

- STAR https://github.com/alexdobin/STAR **(see above)**
- samtools http://www.htslib.org/download/ **(see above)**
- possibly some non-standard Perl modules - see below:
```
A typical perl module installation may involve:
   perl -MCPAN -e shell
   install DB_File
   install URI::Escape
   install Set::IntervalTree
   install Carp::Assert
   install JSON::XS
   install PerlIO::gzip
```

### FusionInspector
FusionInspector is a component of the Trinity Cancer Transcriptome Analysis Toolkit, and as with the other fusion-transcriptome components of CTAT, FusionInspector leverages the CTAT genome lib data resource, containing a reference genome and corresponding protein-coding gene annotation set, and various metadata.
https://github.com/FusionInspector/FusionInspector/wiki/installing-FusionInspector
License: Broad (allowing redistribution).

Requires:

- STAR - currently using version 2.6.1a **(note Arriba wants newer)**
you can use a newer version of STAR, but it may not be compatible with the plug-n-play data resource lib described below.
- samtools (modern versions, 1.3+) **(see above)**
- Perl module: Set::IntervalTree
- Python modules: igv-reports, requests
- Trinity (optional but recommended)
- GMAP (optional but recommended)

### TrinityFusion
Fusion and foreign transcript detection via RNA-seq de novo assembly
https://github.com/trinityrnaseq/TrinityFusion/wiki
License: trinityrnaseq (allowing redistribution)

Requires:
- The Trinity assembler https://github.com/trinityrnaseq/trinityrnaseq/releases
- GMAP version 2017-11-15 http://research-pub.gene.com/gmap/src/gmap-gsnap-2017-11-15.tar.gz
- Bowtie2 aligner https://sourceforge.net/projects/bowtie-bio/files/bowtie2/2.3.3.1/bowtie2-2.3.3.1-linux-x86_64.zip/download **(we have earlier version 2.2.5)**

Note that Trinity also requires:
- jellyfish http://www.genome.umd.edu/jellyfish.html

Some of these have Singularity containers, not sure if anything has a full pipeline with all of these.

**Comment by heatherkellyucl** [comment]
Note: EasyBuild has Arriba, STAR-Fusion, Trinity
Does not have STAR-SEQR, FusionInspector, TrinityFusion (unless wrapped into one of the above).

**Comment by heatherkellyucl** [comment]
Checking what and how much of this is still required as it has been a long time.

**Comment by heatherkellyucl** [comment]
No longer needed, closing. (If anyone reading this later does need this, or parts of this, let us know at rc-support).
